src/pages/api/auth-connect/send-identity-validation-code.ts

```diff
@@ -33,84 +33,87 @@ import { getIdentityType } from "@src/utils/get-identity-type";
+
+            let temporaryIdentity: TemporaryIdentity = {
+              eventId: eventId,
+              value: identityInput.value,
+              type: identityInput.type,
+              expiresAt: identityInput.expiresAt,
+              primary: identityInput.primary,
             };
-          }
 
-          await insertTemporaryIdentity(userCookie.sub, temporaryIdentity);
+            if (identityToUpdateId) {
+              temporaryIdentity = {
+                ...temporaryIdentity,
+                identityToUpdateId,
+              };
+            }
 
-          const verificationCodeMessage =
-            getIdentityType(identityInput.type) === IdentityTypes.EMAIL
-              ? "Confirmation email has been sent"
-              : "Confirmation SMS has been sent";
+            await insertTemporaryIdentity(userCookie.sub, temporaryIdentity);
 
-          setAlertMessagesCookie(response, verificationCodeMessage);
+            const verificationCodeMessage =
+              getIdentityType(identityInput.type) === IdentityTypes.EMAIL
+                ? "Confirmation email has been sent"
+                : "Confirmation SMS has been sent";
 
-          response.statusCode = HttpStatus.OK;
-          response.setHeader("Content-Type", "application/json");
-          response.json({ eventId });
-          return;
-        })
-        .catch((error) => {
-          if (
-            error instanceof IdentityAlreadyUsedError ||
-            error instanceof IdentityValueCantBeBlankError
-          ) {
-            span.setDisclosedAttribute(
-              "Identity input exception",
-              error.message,
-            );
-
-            response.statusCode = HttpStatus.BAD_REQUEST;
-            response.json({ errorMessage: error.message });
+            setAlertMessagesCookie(response, verificationCodeMessage);
+
+            response.statusCode = HttpStatus.OK;
+            response.setHeader("Content-Type", "application/json");
+            response.json({ eventId });
             return;
-          }
-
-          response.statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
-          response.end();
-        });
-    }
-
-    response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
-    response.setHeader("location", "/");
-    response.end();
-    return;
-  });
+          })
+          .catch((error) => {
+            if (
+              error instanceof IdentityAlreadyUsedError ||
+              error instanceof IdentityValueCantBeBlankError
+            ) {
+              span.setDisclosedAttribute(
+                "Identity input exception",
+                error.message,
+              );
+
+              response.statusCode = HttpStatus.BAD_REQUEST;
+              response.json({ errorMessage: error.message });
+              return;
+            }
+
+            response.statusCode = HttpStatus.INTERNAL_SERVER_ERROR;
+            response.end();
+          });
+      }
```

Comment:
Why didn't you do the same here as in `re-send-validation-code.ts` ? I think it's the same, no ?

Reply:
I did nothing to this because I did not want to create conflicts with your PR. Do you want me to handle those?

Reply:
Nope, you're right, I'll do it in mine ;)